app/models/pregnancy.rb

```diff
@@ -33,6 +33,15 @@ class Pregnancy
   # Validations
   validates :created_by,
             presence: true
+  before_update :check_other_validations?
+
+  def check_other_validations?
+    dcaf_soft_pledge.present? &&
+    Patient.first.name.present? &&
```

Comment:
Swap these out for lowercase `patient` and remove the `first`s -- it'll be smart enough to pull the associated object thru activerecord.